src/osd/pg_scrubber.cc

```diff
@@ -616,6 +624,8 @@ bool PgScrubber::range_intersects_scrub(const hobject_t& start, const hobject_t&
  */
 void PgScrubber::add_delayed_scheduling()
 {
+  m_end = m_start; // not blocking any range now
```

Comment:
I don't understand this one.  Perhaps this should be set when we complete a chunk?

Reply:
Yes to your question: this is what we would have done if we were able to complete the chunk. But as we were preempted:

There are two paths for preemption: 

1) locally, which means we are either in state BuildMap or in WaitReplicas:
We won't be scrubbing anything, but we still maintain the object range.
Setting 'end' to 'start' below (l. 591) frees that block. But it's also solved by the other change - we no longer block preemption attempt
(but that path is slower).

2) a replica might signal in its returning message that *it* was preempted (and will not be providing a map).
The primary may still be working on creating its own map. 
But then we notice the preemption, and while we
(unlike the original code) do not retry selecting a range immediately (we wait for all replicas, then spend time requeuing
vie PendingTimer), at least we allow multiple preemptions.
And - once we get to PendingTimer - there is no reason for us to maintain our hold on the old chunk

Reply:
> I don't understand this one. Perhaps this should be set when we complete a chunk?

m_start = m_end would be to complete a chunk

m_end = m_start when aborting/preempting the current chunk to stop blocking any range [m_start, m_end)